udata/core/spam/models.py

```diff
@@ -0,0 +1,193 @@
+from flask import current_app
+from langdetect import detect
+
+from udata.models import db
+from .signals import on_new_potential_spam
+
+NOT_CHECKED = 'not_checked'
+POTENTIAL_SPAM = 'potential_spam'
+NO_SPAM = 'no_spam'
+
+SPAM_STATUS_CHOICES = [NOT_CHECKED, POTENTIAL_SPAM, NO_SPAM]
+
+
+class SpamInfo(db.EmbeddedDocument):
+    status = db.StringField(choices=SPAM_STATUS_CHOICES, default=NOT_CHECKED)
+    callbacks = db.DictField(default={})
+
+
+class SpamMixin(object):
+    spam = db.EmbeddedDocumentField(SpamInfo)
+
+    attributes_before = None
+    detect_spam_enabled: bool = True
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        # Save the list of texts at initialisation to not recheck for spam a text
+        # if it didn't change during this request lifecycle.
+        self.attributes_before = self.texts_to_check_for_spam()
+
+    @staticmethod
+    def spam_words():
+        return current_app.config.get('SPAM_WORDS', [])
```

Comment:
For an exhaustive view of all settings available, please add the default value for new configs in `settings.py`.